compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -676,7 +676,11 @@ impl<'a, 'tcx> MirUsedCollector<'a, 'tcx> {
         &mut self,
         limit: Limit,
         operand: &mir::Operand<'tcx>,
+        allow_move: bool,
     ) -> Option<Size> {
+        if allow_move && let mir::Operand::Move(_) = operand {
```

Comment:
Check this at the one call site instead of passing a bool arg.

Also add a comment explaining this check